forEacher.go

```diff
@@ -0,0 +1,208 @@
+package mrT
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+
+	"github.com/itsmontoya/middleware"
+	"github.com/missionMeteora/journaler"
+	"github.com/missionMeteora/uuid"
+)
+
+func newForEacher(tid string, fn ForEachFn, mw *middleware.MWs, cor bool) *forEacher {
+	var fe forEacher
+	fe.tid = tid
+	fe.fn = fn
+	fe.mw = mw
+	fe.cor = cor
+
+	if tid == "" {
+		fe.state = 1
```

Comment:
Hmm could we add some comments here? Sort of confusing as to what tid exactly is and what each state int represents